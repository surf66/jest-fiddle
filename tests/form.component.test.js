import React from 'react';
import Form from '../src/js/components/form.component.js';
import renderer from 'react-test-renderer';

test('Form component snapshot test', () => {
  const tree = shallow(<Form />);

  expect(tree).toMatchSnapshot();
});

test('should update note in component state when text field changed', () => {
  const component = shallow(<Form />);
  const mockedEvent = { target: { value: 'my new note' } }
  
  component.find('input').simulate('change', mockedEvent);

  expect(component.instance().state.note).toEqual('my new note');
});

test('should call onSubmit callback when form is submitted', () => {
  const onSubmit = jest.fn();
  const component = shallow(<Form onSubmit={onSubmit} />);
  const mockedEvent = { preventDefault: () => {} }

  component.instance().setState({ note: 'my note' });
  component.find('form').simulate('submit', mockedEvent);

  expect(onSubmit).toHaveBeenCalledWith('my note');
});