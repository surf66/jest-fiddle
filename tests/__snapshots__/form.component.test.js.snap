// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component snapshot test 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button>
    Add note
  </button>
</form>
`;
